Expose endpoint for running tests & add cron job to run tests periodically

jhadvig opened the pull request:
Here are the changes for running tests after hitting pods `/protractor` endpoint. Added cronjob for running the tests every 5 minutes.
To test you have to rebuild the image with latest changes and push to dockerhub. Wasnt sure whats the versioning schema so put it to latest.
The whole list is in `kube/smoke-test-list.yaml`

@benjaminapetersen PTAL

jhadvig commented on test/protractor.conf.js:
```diff
@@ -21,7 +25,7 @@ const screenshotReporter = new HtmlScreenshotReporter({
   consolidateAll: true,
   savePath: `${output}/junit`,
-  filePrefix: 'e2e-results'
+  filePrefix: `${timeStamp}-e2e-results`
```
Here I wasnt really sure if we want to keep only the last test result, or we wanna all of them, so I've added timestamp to the filename. Although now I think it would be better to just keep the latest and since we dont intend to store the results anywhere. Instead we could set the result scraping after the tests are done ?
Thoughts ?

benjaminapetersen replied:
Yeah, I've been thinking about that too.  Def looking forward to spending more time fiddling with Prometheus itself & seeing how these metrics get aggregated.  

We could always use an environment var to timestamp them, but default to not using the timestamp.  How do you feel about that?

jhadvig replied:
yeah so the results are relatively small ATM but not sure what we end up with. Safe way would be to store just the latest and set the cron to run the test every new hour and schedule the scrapping after that.

benjaminapetersen replied:
Makes sense to me!

benjaminapetersen commented on kube/smoke-test-list.yaml:
```diff
@@ -0,0 +1,64 @@
+    selector:
+      app: origin-web-console-smoke-test
+    type: ClusterIP
+    clusterIP: 172.30.136.123
```
This looks good.  Curious about `clusterIP`, I'm guessing this is one item we may have to update before deployment each time, similar to the above `CONSOLE_URL`?

jhadvig replied:
so the `clusterIP` will set the IP for the service but if we use it for online, there is a good chance that its going to be used... will probably decouple the pod+service and the cronJob. Put the service+pod into the `pods/smoke-test.yaml` and create a separate `jobs/smoke-test.yaml`

benjaminapetersen commented on kube/smoke-test-list.yaml:
```diff
@@ -0,0 +1,64 @@
+            containers:
+            - name: run-protractor
+              image: byrnedo/alpine-curl
+              command: ["/bin/sh",  "-c", "curl -k 172.30.136.123:3000/protractor"]
```
Should we comment this as well, just to make sure the IPs match as we edit?  

Its prob not a huge deal, just thinking as I've had to `cluster down` and `cluster up` whenever I go to work or go home because my machine IP changes.

jhadvig replied:
Agreed, will add the comment :+1:

benjaminapetersen commented on test/server.js:
```diff
@@ -60,5 +60,23 @@ server.get('/metrics', (req, res) => {
+  });
+  res.end();
+});
+
```
Nice! Shall we `[info]` prefix these?  I dunno how much we care about consistency for this.  I did the same with the `/helpers/logger.js` [logger](https://github.com/benjaminapetersen/origin-web-console-smoke-test/pull/12/files#diff-1d35ece61a2a4349acf21a9db5efa932) & added a little color...prob the color was unnecessary, heh.

jhadvig replied:
yeah lets prefix them. Although not sure if we want to he in upper case `[INFO]` or lowercase `[info]`.. since the `lib/log/output.sh` is prefixing with the uppercase.

benjaminapetersen replied:
Good catch, i guess match?

jhadvig commented:
@benjaminapetersen I've updated the PR. Added the comments regarding the IP to the service and cronJob in their yaml files. Left the List without the comments.
PTAL

benjaminapetersen commented on protractor.sh:
```diff
@@ -21,28 +21,11 @@ echo 'GET /' | timeout 30 openssl s_client -showcerts -connect "$host:$port" | o
   exit $metrics_endpoint_status
 fi
 
-
```
This was kind of fun, sad to see it go :)

benjaminapetersen commented:
@jhadvig merged!  BIG thx for the additions.  If there is any little additional instructions for using it (gotchas?), maybe a quick PR to update the README would be worth a follow.
